### ISSUE TITLE
Where's the logging at?

### ISSUE THREAD

**Issue by tgoodlet** [theelous3/asks#54]
It'd sure be handy if `logging` was used throughout `asks` for debugging purposes 👍

**Comment by theelous3** [comment]
You are absolutely correct.

Clean exceptions and logging are sorely missing for asks at the moment, and are on my todo pile.

**Comment by theelous3** [comment]
There's a fair bit of relevant stuff done above h11 that would be worth logging from the asks level at least. Redirects and so on.

Be warned before you go looking. Some of this lib is absolute eye bleed and it's a miracle that it does anything. (That's a little higher on the todo pile)

**Comment by tgoodlet** [comment]
@theelous3 yeah for one I'd love to get http packets logged with same beauty as in [`httpie`](https://github.com/jakubroztocil/httpie#161colors-and-formatting). He's using custom [`pygments` formatters](https://github.com/jakubroztocil/httpie/blob/master/httpie/output/formatters/colors.py) which end up with some real slick output in the terminal.

Would you take a PR for this kind of thing?

**Comment by theelous3** [comment]
Yes, that looks lovely.

**Comment by pmav99** [comment]
@tgoodlet @theelous3
Hey guys

When it comes to logging, I think that there is a different scope between apps and libs. 

An application can and should format its logs in the way that makes the most sense. E.g. on an app like `httpie`, which is meant to be used as a console tool and, consequently, log performance is completely irrelevant, using fancy log formatters is definitely the way to go.

On the other had, on a library (e.g. `asks`) all you should be doing is [this](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library), i.e. adding loggers at the module level like this:
``` python
logging.getLogger('foo').addHandler(logging.NullHandler())
```

This way, me, @tgoodlet  and everybody else who uses `asks` will be able to customize the log output to his needs/taste without having to jump through hoops. E.g.:
``` python
logging.getLogger("asks").setLevel(logging.WARNING)
```

Just for reference [this is relevant code](https://github.com/requests/requests/blob/master/requests/__init__.py#L124-L133) from `requests`.
